src/testdrive/src/action/kafka/verify.rs

```diff
@@ -123,6 +135,19 @@ impl Action for VerifyAction {
             }
         };
 
+        if self.nonced_topic {
```

Comment:
I do not think you need the checking code in this block -- if the topic name is not as expected,  then `kafka-verify` will naturally fail anyway because the expected data will not be there.

Reply:
It's helpful to check if the topic ends with a nonce or not (which I missed when working on the consistency piece of `CREATE SINK`). But, I agree that we don't need it for exactly once topics! I removed the `else` clause.